portal-kernel/src/com/liferay/portal/kernel/search/BaseSearchEngine.java

```diff
@@ -44,12 +44,12 @@ public Collection<Long> getIndexedCompanyIds() {
 
 	@Override
 	public IndexSearcher getIndexSearcher() {
-		return _indexSearcher;
+		return new DummyIndexSearcher();
```

Comment:
@W2jx Currently only 2 classes are extending `BaseSearchEngine`, and both are overriding `getIndexSearcher()` and `getIndexWriter()`. So I don't think you need to keep them in `BaseSearchEngine`. Make it an abstract class and don't override these two methods so that it's clear that the sub-classes must implement these methods.

Reply:
Ok, I will fix it.